test/correctness/simd_op_check.cpp

```diff
@@ -1871,6 +1871,16 @@ struct Test {
         check("vmpa(v*.h,r*.b)", hvx_width/2, 2*i32(i16_1) + 3*i32(i16_2));
         check("v*.w += vmpa(v*.h,r*.b)", hvx_width/2, 2*i32(i16_1) + 3*i32(i16_2) + i32_1);
 
+        check("v*:*.h = vtmpy(v*:*.ub, r*.b)", hvx_width/1, 2*i16(in_u8(x + hvx_width - 1)) + 3*i16(in_u8(x + hvx_width)) + i16(in_u8(x + hvx_width + 1)));
```

Comment:
We should also add padding at the max so the original RDom test can work.

This change is sufficient to make simd_op_check work as a correctness test, but the apps/simd_op_check driver also uses this code to generate and actually run the code. That code also needs to be modified the way I described so it can run successfully.

You should try running the simd_op_check app as well, if you haven't already. That verifies that the optimizations actually do what they claim to do (it verifies that the scalar version of the program matches the vector version result). It will also make sure that it still works after adding padding to the buffer we use for simd_op_check.